Update with api changes for v5.0.0 (#398)
cucumber/docs: content/docs/cucumber/configuration.md

# Type Registry

{{% block "java,kotlin" %}}
Parameter types let you convert parameters from cucumber-expressions to objects.

> Data table and doc string string types let you convert data tables and doc

Reviewer: Repeat of "string"; is this intended or not? If so, maybe put `doc string` in back tics to indicate it is terminology.

strings to objects. Like step definitions, type definitions are part of the glue.

> When placed on the glue path Cucumber will detect them automatically. 

Reviewer: Do we explain "glue" and "glue path" anywhere? Am thinking a glossary might be helpful (probably outside of the scope of this PR)

{{% /block %}}

{{% block "java" %}}

> For example, the following class registers custom "Author" data table type:

Reviewer: ```suggestion
For example, the following class registers a custom "Author" data table type:
```

        
        DataTableType((Map<String, String> entry) -> new Author(
            entry.get("firstName"),
            entry.get("lastName"),
            entry.get("famousBook")
        ));

        ParameterType("book", ".*", (String bookName) -> new Book(bookName));

        DocStringType("json", (String docString) -> 
            objectMapper.readValue(docString, JsonNode.class));
    }
}
```

{{% /block %}}

{{% block "kotlin" %}}

```kotlin
package com.example

import com.fasterxml.jackson.databind.JsonNode;
import com.fasterxml.jackson.databind.ObjectMapper;

import io.cucumber.java8.En

class LambdaStepDefinitions : En {

    init {
        val objectMapper = ObjectMapper()

        ParameterType("book", ".*") { s : String ->
            Book(s)
        }

        DataTableType { entry: Map<String, String> ->
            Author(entry["firstName"], entry["lastName"], entry["famousBook"])
        }

        DocStringType("json") { docString: String -> 
	        objectMapper.readValue(docString, JsonNode::class)
	    }
    }
}
```
{{% /block %}}

{{% block "java,kotlin" %}}
Using the `@DefaultParameterTransformer`, `@DefaultDataTableEntryTransformer`

> and `DefaultDataTableCellTransformer` annotations also possible to plugin an 

Reviewer: ```suggestion
and `@DefaultDataTableCellTransformer` annotations, it is also possible to plug in an 
```

public class StepDefinitions {

    private final ObjectMapper objectMapper = new ObjectMapper();

    @DefaultParameterTransformer
    @DefaultDataTableEntryTransformer
    @DefaultDataTableCellTransformer
    public Object transformer(Object fromValue, Type toValueType) {
        return objectMapper.convertValue(fromValue, objectMapper.constructType(toValueType));
    }
}
```
{{% /block %}}

{{% block "kotlin" %}}
```kotlin
package com.example

import com.fasterxml.jackson.databind.ObjectMapper
import io.cucumber.java.DefaultDataTableCellTransformer
import io.cucumber.java.DefaultDataTableEntryTransformer
import io.cucumber.java.DefaultParameterTransformer

import java.lang.reflect.Type

class StepDefinitions {

    private val objectMapper = ObjectMapper()

    @DefaultParameterTransformer
    @DefaultDataTableEntryTransformer
    @DefaultDataTableCellTransformer
    fun transformer(fromValue: Any, toValueType: Type): Any {
        return objectMapper.convertValue(fromValue, objectMapper.constructType(toValueType))
    }
}
```
{{% /block %}}

{{% block "java,kotlin" %}}
For lambda defined step definitions, there are `DefaultParameterTransformer`, 

> `DefaultDataTableCellTransformer` and `DefaultDataTableEntryTransformer` functions:

Reviewer: ```suggestion
`DefaultDataTableCellTransformer` and `DefaultDataTableEntryTransformer` {{% stepdef-body %}}:
```

Reviewer: Use this short code to correctly display "method" or "function" for the respective programming language. (Although I'm not sure if/how that would work with a text for 2 different languages)

{{% /block %}}

{{% block "java,kotlin,ruby" %}}

> If you are using a type that has not yet been defined, you will got an error similar to:

Reviewer: ```suggestion
If you are using a type that has not yet been defined, you will get an error similar to:
```

Reviewer: Accidental change?
